File: AplicacionWeb/Scripts/Usuario.js
/// <reference path="jquery-2.0.0-vsdoc.js" />
$(document).ready(function (e) {

    if ($("#IDUsuario").val() != "" && $("#opcURL").val() == "editar") {
        $(".pagetitle h2").html("Datos de Usuario");
        $("#nuevo").parent("li").hide();
        $("#guardar_nuevo").parent("li").hide();
        ObtenerDatosDeUsuario();
    }
    DesmarcarCamposVaciosLLenados(); //desmarcamos los campos marcados que se van llenando
    //    $("input[name=Bloquear]:radio").click(function (e) {
    //        e.preventDefault();
    //        alert($(this).val());
    //    });

    $("#nuevo").click(function (e) {
        e.preventDefault();
        //alert("kk");
        $("#IDUsuario").val("");
        limpiarMsjBox();
        limpiarTodosLosCampos();
    });
    $("#guardar").click(function (e) {
        e.preventDefault();
        if (CamposRequeridosVacios(".requerido")) {//si estan vacios
            showError("Los campos marcados son obligatorios");
            return false;
        } else {
            limpiarMsjBox();
            quitarBgerror();
            if ($("#contrasenia1").val() != "" || $("#contrasenia2").val() != "") {
                if (ValidarContrasenias("#contrasenia1", "#contrasenia2") == true) {
                    $("input[type='password']").val("");
                    return false;
                }
            }
            if (GuardarUsuario() > 0) {
                ObtenerDatosDeUsuario();
                $("input[type='password']").val("");
            }
        }
    });
    $("#guardar_cerrar").click(function (e) {
        e.preventDefault();
        if (CamposRequeridosVacios(".requerido")) {//si estan vacios
            showError("Los campos marcados son obligatorios");
            return false;
        } else {
            limpiarMsjBox();
            quitarBgerror();
            if ($("#contrasenia1").val() != "" || $("#contrasenia2").val() != "") {
                if (ValidarContrasenias("#contrasenia1", "#contrasenia2") == true) {
                    $("input[type='password']").val("");
                    return false;
                }
            }
            if (GuardarUsuario() > 0) {
                location.href = "GestionUsuarios.aspx";
            }
        }
    });
    $("#guardar_nuevo").click(function (e) {
        e.preventDefault();
        if (CamposRequeridosVacios(".requerido")) {//si estan vacios
            showError("Los campos marcados son obligatorios");
            return false;
        } else {
            limpiarMsjBox();
            quitarBgerror();
            if ($("#contrasenia1").val() != "" || $("#contrasenia2").val() != "") {
                if (ValidarContrasenias("#contrasenia1", "#contrasenia2") == true) {
                    $("input[type='password']").val("");
                    return false;
                }
            }
            if (GuardarUsuario() > 0) {
                limpiarTodosLosCampos();
            }
        }
    });

});           //fin document


//----------funciones de usuario
function GuardarUsuario() {
    var ID = 0;
    var usuario = new Object();
    usuario.USUA_Interno=$("#IDUsuario").val();
    usuario.USUA_Apellido=$("#Apellido").val();
    usuario.USUA_Nombre = $("#Nombre").val();
    usuario.USUA_Usuario = $("#Usuario").val();
    usuario.USUA_Activo = $("input[name='Bloquear']:checked").val();
    usuario.USUA_Direccion=$("#Direccion").val();
    usuario.USUA_Correo=$("#Email").val();
    usuario.GRUP_Interno = $("#Grupo").val();
    usuario.USUA_Contrasenia = $("#contrasenia1").val();

    $.ajax({
        data: JSON.stringify(usuario),
        type: "POST",
        dataType: "json",
        contentType: 'application/json; charset=utf-8',
        url: "UsuarioDatos.aspx/GuardarUsuario",
        async: false,
        cache: false,
        success: function (data) {
            ID = parseInt(data.d);

            if (ID > 0) {
                showMensaje("Datos guardados correctamente");
                $("#IDUsuario").val(ID);
                
            } else {
                showError("No se insertó el usuario");
                //$("#IDUsuario").val("");
            }
        } //fin success
    }); //fin de ajax
    return ID;
} //fin guardar usuario
function ObtenerDatosDeUsuario() {
    var usuario = new Object();
    usuario.USUA_Interno = $("#IDUsuario").val();
    $.ajax({
        data: JSON.stringify(usuario),
        type: "POST",
        dataType: "json",
        contentType: 'application/json; charset=utf-8',
        url: "UsuarioDatos.aspx/ObtenerUsuario",
        async: false,
        cache: false,
        success: function (data) {
            var item = data.d;

            $("#IDUsuario").val(item.USUA_Interno);
            $("#Apellido").val(item.USUA_Apellido);
            $("#Nombre").val(item.USUA_Nombre);
            $("#Usuario").val(item.USUA_Usuario);

            if (item.USUA_Activo == true)
                $("input[name='Bloquear'][value='true']").attr("checked", "checked");
            else
                $("input[name='Bloquear'][value='false']").attr("checked", "checked");
            $("#Direccion").val(item.USUA_Direccion);
            $("#Email").val(item.USUA_Correo);
            $("#Grupo").val(item.GRUP_Interno);
            $("#UsuarioCreador").val(item.UsuarioCreador);

            if (item.HIIN_FechaIngreso != null)
                $("#FechaUltimoIngreso").val(ConvertirDateTimeAString(item.HIIN_FechaIngreso, "dd/mm/yyyy HH:MM:ss.l"));
            else
                $("#FechaUltimoIngreso").val("");

            if (item.AUDI_FechaCrea != null)
                $("#FechaRegistro").val(ConvertirDateTimeAString(item.AUDI_FechaCrea, "dd/mm/yyyy HH:MM:ss.l"));
            else
                $("#FechaRegistro").val("");

            $("#IPacceso").val(item.HIIN_IPacceso);
        }
    });    //fin de ajax
}
function ValidarContrasenias(input1, input2) {
    var error = false;
    if ($("#IDUsuario").val() == "") {
        if ($(input1).val() == "" || $(input2).val() == "") {
            showError("Escriba una contraseña. Mínimo 6 dígitos");
            return true;
        }
    }
    if ($(input1).val() != $(input2).val()) {
        error = true;
        showError("Las contraseñas no coinciden. Por favor vuelva a escribirlas. Se hace diferencia entre mayúscula y minúscula");
    } else if ($(input1).val().length < 6) {
        error = true;
        showError("La contrasenia no debe tener menos de 6 dígitos");
    }
    return error;
}

File: MantenimientoEntidades/Usuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Mantenimiento.Entidades
{
    public class Usuario
    {
        #region Propiedades
        private int? _USUA_Interno;
        public int? USUA_Interno
        {
            get { return _USUA_Interno; }
            set { _USUA_Interno = value; }
        }

        private string _USUA_Usuario;
        public string USUA_Usuario
        {
            get { return _USUA_Usuario; }
            set { _USUA_Usuario = value; }
        }

        private string _USUA_Contrasenia;
        public string USUA_Contrasenia
        {
            get { return _USUA_Contrasenia; }
            set { _USUA_Contrasenia = value; }
        }

        private string _USUA_Nombre;
        public string USUA_Nombre
        {
            get { return _USUA_Nombre; }
            set { _USUA_Nombre = value; }
        }

        private string _USUA_Apellido;
        public string USUA_Apellido
        {
            get { return _USUA_Apellido; }
            set { _USUA_Apellido = value; }
        }

        private string _USUA_Direccion;
        public string USUA_Direccion
        {
            get { return _USUA_Direccion; }
            set { _USUA_Direccion = value; }
        }

        private string _USUA_Correo;
        public string USUA_Correo
        {
            get { return _USUA_Correo; }
            set { _USUA_Correo = value; }
        }

        private Boolean _USUA_Activo;
        public Boolean USUA_Activo
        {
            get { return _USUA_Activo; }
            set { _USUA_Activo = value; }
        }

        private int? _GRUP_Interno;
        public int? GRUP_Interno
        {
            get { return _GRUP_Interno; }
            set { _GRUP_Interno = value; }
        }

        private string _GRUP_Nombre;
        public string GRUP_Nombre
        {
            get { return _GRUP_Nombre; }
            set { _GRUP_Nombre = value; }
        }

        private int _AUDI_UsuarioCrea;
        public int AUDI_UsuarioCrea
        {
            get { return _AUDI_UsuarioCrea; }
            set { _AUDI_UsuarioCrea = value; }
        }

        private string _UsuarioCreador;
        public string UsuarioCreador
        {
            get { return _UsuarioCreador; }
            set { _UsuarioCreador = value; }
        }

        private DateTime _AUDI_FechaCrea;
        public DateTime AUDI_FechaCrea
        {
            get { return _AUDI_FechaCrea; }
            set { _AUDI_FechaCrea = value; }
        }

        private DateTime? _HIIN_FechaIngreso;//Para el ultimo ingreso del usuario
        public DateTime? HIIN_FechaIngreso
        {
            get { return _HIIN_FechaIngreso; }
            set { _HIIN_FechaIngreso = value; }
        }

        private string _HIIN_IPacceso;
        public string HIIN_IPacceso
        {
            get { return _HIIN_IPacceso; }
            set { _HIIN_IPacceso = value; }
        }

        private string _USUA_ID;
        public string USUA_ID
        {
            get { return _USUA_ID; }
            set { _USUA_ID = value; }
        }


        #endregion
        #region Contructores
        public Usuario()
        {
        }
        public Usuario(int? USUA_Interno, string USUA_Usuario, string USUA_Nombre, string USUA_Apellido, string USUA_Direccion, string USUA_Correo, Boolean USUA_Activo, int? GRUP_Interno)
        {
            this.USUA_Interno = GRUP_Interno;
            this.USUA_Usuario = USUA_Usuario;
            this.USUA_Nombre = USUA_Nombre;
            this.USUA_Apellido = USUA_Apellido;
            this.USUA_Direccion = USUA_Direccion;
            this.USUA_Correo = USUA_Correo;
            this.USUA_Activo = USUA_Activo;
            this.GRUP_Interno = GRUP_Interno;
        }
        #endregion
    }
}
